core/commit.go

```diff
@@ -133,6 +140,7 @@ func makeCommitmentCounter(f uint32) commitmentCounter {
 
 	return func(replicaID uint32, prepare *messages.Prepare) (done bool, err error) {
 		primaryID := prepare.ReplicaID()
+		prepareView := prepare.View()
```

Comment:
`primaryID` is always equal to `prepareView % n`, but if that's not the case, the message should be suspected seriously. So is it worth asserting that?

Reply:
As documented in definition of `commitmentCounter` type, the passed Prepare message is supposed to be validated for internal consistency. This means `prepare.ReplicaID()` must denote the primary replica of view `prepare.View()` in the current configuration.